inspirehep/modules/relations/migrate.py

```diff
@@ -0,0 +1,258 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+from functools import partial
+
+import inspirehep.modules.relations.command_producers as commands
+from inspirehep.modules.relations import builders
+from inspirehep.modules.relations.graph_representation import (
+    get_central_node,
+    get_node_labels,
+    get_node_properties,
+    get_node_uid,
+    get_node_type,
+    get_outgoing_relations,
+    get_relation_type,
+    get_start_node,
+    get_end_node,
+    produce_model_from_node,
+    should_create_end_node_if_not_exists
+)
+
+from inspirehep.modules.pidstore.providers import InspireRecordIdProvider
+
+from uuid import uuid4
+
+import csv
+
+import os
+
+import shutil
+
+
+INDEX_TO_BUILDER = {
+    'authors': builders.hepnames.hepnames,
+    'journals': builders.journals.journals,
+    'jobs': builders.jobs.jobs,
+    'literature': builders.literature.literature,
+    'experiments': builders.experiments.experiments,
+    'conferences': builders.conferences.conferences,
+    'institutions': builders.institutions.institutions
+}
+
+
+def nested_setdefault(dictionary, keys, default=None):
+    current_dict = dictionary
+    for key in keys[:-1]:
+        try:
+            current_dict = current_dict.setdefault(key, dict())
+        except AttributeError:
+            raise TypeError(
+                'Wrong internal structure. ' + '.'.join(
+                    keys[:keys.index(key)]
+                    ) + ' is already something else than dict.'
+                )
+    return current_dict.setdefault(keys[-1], default)
```

Comment:
If you have:
```python
keys = ['first_level', 'second_level', 'third']
```

and you call:
```python
nested_setdefault(my_dict, keys)
```

then you want `nested_setdefualt` to do basically this:

```python
my_dict['first_level']['second_level'].setdefault('third', default)
```
That's why `my_dict`, `my_dict['first_level']` and `my_dict['second_level']` need to be dictionaries.

I'll add documentation before I leave, sorry it's not here yet ;)